### PR TITLE
Increment version for spring release

### PR DESCRIPTION
**Pull request by chenrujun**
Prepare to release SpringCloudAzure:3.14.0

### CODE REVIEW

File: sdk/boms/azure-spring-boot-bom/pom.xml [hunk 1]
```diff
@@ -37,11 +37,11 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <packageOutputDirectory>${project.build.directory}</packageOutputDirectory>
     <legal><![CDATA[[INFO] Any downloads listed may be third party software.  Microsoft grants you no rights for third party software.]]></legal>
-    <azure.spring.boot.version>3.12.0</azure.spring.boot.version>
```
Comment (chenrujun) [inline sdk/boms/azure-spring-boot-bom/pom.xml:40]: @backwind1233 , seems it is an error in last release?

**Comment by saragluna** [comment]
@chenrujun, we should hold the release for the listener error handler feature. @yiliuTo when is the time we promise the customer about the error handler?

**Comment by yiliuTo** [comment]
@chenrujun @saragluna , the ETA of error handler is the end of Febuary.

**Comment by chenrujun** [comment]
I'll close this PR and delete [release/spring-cloud-azure_3.14](https://github.com/Azure/azure-sdk-for-java/tree/release/spring-cloud-azure_3.14) branch. Because this branch is a hint of `checkpoint`. I'll create a new branch from main branch to do release.

**Pull request closed without merging**